src/Stratis.Bitcoin/P2P/Peer/NetworkPeerCollection.cs

```diff
@@ -103,16 +103,16 @@ public INetworkPeer FindByEndpoint(IPEndPoint endpoint)
 
         private static bool Match(IPAddress ip, int? port, INetworkPeer peer)
         {
-            if (port.HasValue)
-            {
-                return ((peer.State == NetworkPeerState.Connected || peer.State == NetworkPeerState.HandShaked) && peer.RemoteSocketAddress.Equals(ip) && 
-                        (peer.RemoteSocketPort == port.Value)) || (peer.PeerVersion.AddressFrom.Address.Equals(ip) && (peer.PeerVersion.AddressFrom.Port == port.Value));
-            }
-            else
-            {
-                return ((peer.State == NetworkPeerState.Connected || peer.State == NetworkPeerState.HandShaked) && 
-                        peer.RemoteSocketAddress.Equals(ip)) || peer.PeerVersion.AddressFrom.Address.Equals(ip);
-            }
+            bool isConnectedOrHandShaked = (peer.State == NetworkPeerState.Connected || peer.State == NetworkPeerState.HandShaked);
+
+            bool isAddressMatching = peer.RemoteSocketAddress.Equals(ip) 
+                                     && (!port.HasValue || port == peer.RemoteSocketPort);
+
+            bool isPeerVersionAddressMatching = peer.PeerVersion?.AddressFrom != null
+                                                && peer.PeerVersion.AddressFrom.Address.Equals(ip)
+                                                && (!port.HasValue || port == peer.PeerVersion.AddressFrom.Port);
```

Comment:
Does this get disabled if its false?
` && (!port.HasValue || port == peer.PeerVersion.AddressFrom.Port)`

Reply:
as discussed you can read it as "if port has no value, or if part has a value and that value matches" :)